project_status.py

```diff
@@ -215,108 +211,449 @@ def parse_qualimap(genome_results_file, sample):
+    result = {}
+    for sample_id, sample in results_francesco.items():
+        sequenced = sample.get('#Archived_runs', '')
+        organized = sample.get('#Data_runs', '')
+        analyzed = sample.get('#Analysis_runs', '')
+        try:
+            sequenced = int(sequenced)
+            organized = int(organized)
+            analyzed = int(analyzed)
+        # if not int or something strange in the results, print it too (just in case)
+        except ValueError, e:
+            result[sample_id] = {'sequenced': sequenced, 'organized': organized, 'analyzed': analyzed}
+        else:
+            if not(sequenced == organized == analyzed):
+                result[sample_id] = {'sequenced': sequenced, 'organized': organized, 'analyzed': analyzed}
+    return result
+
+
+def get_sequenced(project):
+    incoming = "/proj/ngi2016003/incoming"
+    project_flowcells = {}
+    for fc in os.listdir(incoming):
+        sample_sheet = os.path.join(incoming, fc, 'SampleSheet.csv')
+        command = 'grep {} {}'.format(project, sample_sheet)
+        try:
+            p = subprocess.Popen(command, shell=True, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        except Exception, e:
+            print 'Command failed: {}'.format(command)
+            raise e
+        output = p.communicate()[0]
+        if output:
+            for line in output.split('\n'):
+                # if not an empty line
+                if line:
+                    try:
+                        sample = line.split(',')[2]
+                    except Exception, e:
+                        print line
+                        print 'Skipping line: {} from sample sheet: {}'.format(line, sample_sheet)
+                        # if something went wrong
+                        continue
+                    else:
+                        if sample not in project_flowcells:
+                            project_flowcells[sample] = [fc]
+                        else:
+                            project_flowcells[sample].append(fc)
+    return project_flowcells
+
+def get_organized(project):
+    sequenced = get_sequenced(project)
+    project_path = os.path.join('/proj/ngi2016003/nobackup/NGI/DATA', project)
+    organized = {}
+    for sample in sequenced:
+        for fc in sequenced[sample]:
+            # '*'' is libprep, can be 'A', 'B', etc
+            path = os.path.join(project_path, sample, '*', fc)
+            if glob.glob(path): # list of files
+                if sample not in organized:
+                    organized[sample] = [fc]
+                elif fc not in organized[sample]:
+                    organized[sample].append(fc)
+                # else: skip -> we don't want duplicates
+    return organized
+
+def get_reprepped(project):
+    pass
+
+def get_not_organized(project):
+    flowcells_samples = get_sequenced(project)
+    project_path = os.path.join('/proj/ngi2016003/nobackup/NGI/DATA', project)
```

Comment:
you need to make this less verbose and allow different uppmax projects to be specified